libc/src/stdlib/atexit.cpp

```diff
@@ -71,20 +67,31 @@ void call_exit_callbacks() {
   ExitCallbackList::destroy(&exit_callbacks);
 }
 
-} // namespace internal
-
-static int add_atexit_unit(const AtExitUnit &unit) {
+int add_atexit_unit(const AtExitUnit &unit) {
   MutexLock lock(&handler_list_mtx);
-  if (!exit_callbacks.push_back(unit))
-    return -1;
-  return 0;
+  if (exit_callbacks.push_back(unit))
+    return 0;
+  return -1;
 }
 
+} // namespace
+
+extern "C" {
+
 // TODO: Handle the last dso handle argument.
-extern "C" int __cxa_atexit(AtExitCallback *callback, void *payload, void *) {
+int __cxa_atexit(AtExitCallback *callback, void *payload, void *) {
```

Comment:
is there a reason this shouldn't be `extern "C"`?

Reply:
It still is, I just changed the code to use an `extern "C"` block (see line 79) rather than declare BOTH `__cxa_atexit` and `__cxa_finalize` with `extern "C"` on their function signatures.

Reply:
ah, I missed that.